File: 0151-reverse-words-in-a-string/0151-reverse-words-in-a-string.cpp
class Solution {
public:
    string reverseWords(string s) {
       int n = s.size();
       string ans = "";
       reverse(s.begin(), s.end());    // Reverse entire given string

       for(int i = 0; i < n; i++){
        string word = "";
        while(i < n && s[i] != ' '){    // It checks 1 word at time
            word += s[i];
            i++;
        }

        reverse(word.begin(), word.end()); // word in reverse form, reverse it for correct form

        if(word.size() > 0)     // in case there is a empty space, no need to add space
            ans += " "+ word;

       }
       return ans.substr(1); // it converts into string from the 1st space
    }
};

File: 0151-reverse-words-in-a-string/0151-reverse-words-in-a-string.java
class Solution {
    public String reverseWords(String s) {
        int n = s.length();

        StringBuilder sb = new StringBuilder(s);
        sb.reverse();
        s = sb.toString();

        String ans = "";

        for(int i = 0; i < n; i++){
            String word = "";

            while(i < n && s.charAt(i) != ' '){
                word += s.charAt(i);
                i++;
            }

            StringBuilder w = new StringBuilder(word);
            w.reverse();

            if(word.length() > 0)
                ans += " "+w.toString();

        }
        return ans.substring(1);
    }
}